### ISSUE TITLE
Specific observables section

### ISSUE THREAD

**Issue by dmzuckerman** [dmzuckerman/Sampling-Uncertainty#6]
@mangiapasta and @ajschult - you guys got a great draft going for the specific observables section.  I have some questions/comments that hopefully you could address in the next few days:

- I think this is for Paul: The first subsection addresses 'quality of data', which really means 'overall qualitative behavior of data', I think.  You give an example of material failure with increasing strain.  I think you should point out that this is a single observable measured (I think) over many different simulations.  So we will want a separate example for a single observable measured from a single simulation ... if you put a stub paragraph in for that, I can fill it in.
- Re propagation of uncertainty: I was wondering about the example of block averages for free energy estimation.  If the individual blocks are pathological, isn't that itself an indicator of questionable sampling?  Also, it seems that the later aveat, as well as the subsection on 'synthetic data', calls into question using a Taylor expansion for a nonlinear derived quantity.  So is free energy a good example here?
- Re bootstrapping: Is there an implicit assumption that the original data samples are statistically independent?  Please clarify
- Synthetic data: looks like this should go right after propagation of error since they discuss the same issue if I understand correctly
- Dark uncertainty: Wish I had thought of that name!!

As I said, this is off to a great start.  I'm nitpicking so it can be even better.

**Comment by ajschult** [comment]
Pathologically bad individual blocks aren't a big problem.  If the blocks are small enough, then individual bad blocks are inevitable, but the approach work fine so long as the Taylor series approximation holds.  I would assert that propagation of uncertainty does work well for free energy so long as the calculation as a whole is working.  I have spent a lot of time trying to figure out how to estimate free energy uncertainties where propagation fails, but in the end, I abandoned those free energy methods in favor of alternatives that yielded estimates that were precise enough for propagation to work.  In practice, having a nonlinear example gives an opportunity to point out the limitations.

Bootstrapping -- yes the samples need to be uncorrelated.  I thought I mentioned that, but it looks like not!  I've added a bit.

Synthetic data actually seems a lot like boostrapping, specifically parametric bootstrapping.  In fact, based on the description, I'd have trouble saying how it is different from parametric bootstrapping.  Paul, it'd help if you describe how it is different (or perhaps it is a specific way to bootstrap).

**Comment by agrossfield** [comment]
Re bootstrapping samples needing to be uncorrelated:  You can still do bootstrapping for correlated data.  However, you need to reduce the number of samples you draw to reflect this fact.  If you have 10,000 frames in your trajectory, but your correlation analysis indicate you really only have ~100 independent samples, you can estimate uncertainties by drawing sets of 100 data points.  It's not especially rigorous, but frankly nothing about bootstrapping is rigorous, and in my experience the estimates are sane.  I used this approach in my WHAM code.

**Comment by dmzuckerman** [comment]
@mangiapasta and @ajschult - Andrew, thanks for the further explanation.  I think the fact that the linear/Taylor propagation of error required some discussion among us (the experts!) suggests that could use a little more clarification.  Also, since you are skimming the surface a bit, it would be good to cite references where readers could go for more.

@mangiapasta Paul I just wanted to point you to Andrew's question above in case you missed it about differences between bootstrapping and synthetic data analysis.

@agrossfield also implicitly raised the issue that the correlation time is really a fundamental thing for all to understand.  I will add that to our key definitions

**Comment by mangiapasta** [comment]
I think that the synthetic data approach is in fact parametric bootstrapping.  We just always called in the synthetic data approach.


The main reason why I prefer the parametric approach over others is that one can perform the analysis on correlated data and don't suffer from needing to generate smaller synthetic sets, provided one can model the correlations.  So perhaps it makes sense to merge that section with bootstrapping and emphasize the differences?





________________________________

**Comment by dmzuckerman** [comment]
@mangiapasta Yes, please merge and clarify then.

**Comment by ajschult** [comment]
@mangiapasta we were also doing parametric bootstrapping as well before we knew the method already existed.  I think a merged section will work fine.

**Comment by mangiapasta** [comment]
Just to emphasize that the parametric approach can handle correlated data, cf. appendix in this paper
________________________________

**Comment by dmzuckerman** [comment]
@mangiapasta  Remember if there's anything you need to clarify to us, it certainly needs to be clarified/refereced in manuscript.

**Comment by mangiapasta** [comment]
@dmzuckerman @ajschult Sounds good.  If okay with you two I don't mind taking a first crack at merging / clarifying the boostrap section.  I still need to read over the other sections, so I may have a few more comments.  (Also, apologies for being a bit slow to respond; we're in the midst of an internal proposal exercise that is [thankfully] over today)

**Comment by ajschult** [comment]
@mangiapasta yes, please go ahead with the merge.

**Comment by dmzuckerman** [comment]
@mangiapasta yes, please go ahead.  I would like you and @ajschult to be happy with the section before we turn it over for editing.  I have some ideas for intro material for the section, but I'd like the experts to get the technical stuff straightened.

**Comment by mangiapasta** [comment]
@dmzuckerman Just to reply to some of your original comments:

"The first subsection addresses 'quality of data', which really means 'overall qualitative behavior of data', I think."

Yours is a better way to paraphrase the content of that section.  I'll edit a bit for clarity in light of your comment.  

@dmzuckerman @ajschult Dan's comment, "Synthetic data: looks like this should go right after propagation of error since they discuss the same issue if I understand correctly"

I'm not sure how to handle this comment in light of the previous discussion about connection between synthetic data and bootstrap.  Does this highlight a difference between the two, or can bootstrapping also be used for uncertainty propagation?  I've never used resampling methods for this purpose, and generally in my circle we use synthetic exclusively for the purposes of propagation.

**Comment by dmzuckerman** [comment]
@mangiapasta @ajschult @agrossfield  Doesn't bootstrapping implicitly do error propagation since it provides confidence intervals for derived quantities?  This issue must be in the Tibshirani book.  Do we want to tell readers that bootstrapping/synthetic data are more modern and systematic ways of propagating error compared to linear formulas? Organizationally, the propagation subsection can refer forward to bootstrapping/synthetic as alternatives (or superior methods if that's what you think).

**Comment by ajschult** [comment]
@mangiapasta Yes, bootstrapping can be used for uncertainty propagation.  What I called "simple" bootstrapping can be used when the Taylor series approach fails due to nonlinearity (such as for free energy), which I should probably highlight more clearly.

I consider the Taylor series approach to be the primary means to propagate uncertainty (due to its simplicity), with the caveat that you have to be aware of its limitations.  If we're doing something simple, then we just use the Taylor series.  If we're doing something more complicated and there are signs of trouble (unable to capture scatter in the data, or unable to describe variation when repeating the calculation), then we would (as you say) try other approaches to see which works best.

**Comment by ajschult** [comment]
@dmzuckerman Simple bootstrapping should be more robust than the Taylor series approach.  Parametric bootstrapping has many of the same problems.  It avoids the need to express y=f(x), but can still have trouble with nonlinear functions.  In the free energy example, if <exp(-beta deltaU)> = 1 +/- 0.5 and you generate new estimates for that quantity from a Gaussian centered at 1 with a width of 0.5, then you will at some point generate a negative number that causes catastrophic failure.  You'd need to be aware that the distribution of the average exponentials is not Gaussian and use a different model to generate data (and your results in the might still be sensitive to the choice of model).

**Comment by dmzuckerman** [comment]
@ajschult Thanks for explaining.  Please clarify in the article as needed.

**Comment by ajschult** [comment]
I've tried to address most of what we've talked about here.  I'll need to make a second pass once @mangiapasta folds in the synthetic data section.

**Comment by ajschult** [comment]
I also noticed that we don't have anywhere in the document that discusses repeating simulations to either check or compute uncertainties (except to mention this idea in the definition of precision).  I could add a sentence or two to the block averaging section (it's a lot like using the whole simulation as a block).  Or it could go in a new section.  I see that we originally planned a "basics" section (how to report, what goal to shoot for, significant figures) and it could also live there if we do that.

**Comment by agrossfield** [comment]
Agreed. That's actually far and away the best way to estimate uncertainty (you get to use the classic statistical formulas!), but there is a catch. If you're going to claim the individual trajectories are independent measurements, you need to make the starting structures as different as you can. What that means is system-dependent. For a protein, you need to start with the native state, but you could rebuild the water and environment. If it's a liquid or membrane simulation, you need to rebuild the box with a new random seed.  For materials, there's probably something else you can do.

I'm not going to have a chance to work on this this weekend, which is why I wrote the long paragraph here. Can somebody put something like that in?

Dr. Alan Grossfield
Department of Biochemistry and Biophysics
University of Rochester Medical Center

On Sep 30, 2017, at 9:07 AM, Andrew Schultz <notifications@github.com<mailto:notifications@github.com>> wrote:


I also noticed that we don't have anywhere in the document that discusses repeating simulations to either check or compute uncertainties (except to mention this idea in the definition of precision). I could add a sentence or two to the block averaging section (it's a lot like using the whole simulation as a block). Or it could go in a new section. I see that we originally planned a "basics" section (how to report, what goal to shoot for, significant figures) and it could also live there if we do that.

-

**Comment by ajschult** [comment]
OK, the basics section exists and discusses computing uncertainties from multiple simulations.  The original outline suggested that this section would include "what to shoot for", but I'm not sure how to prescribe a target uncertainty.

**Comment by mangiapasta** [comment]
@ajschult @agrossfield The dark uncertainty method actually does provide a way to estimate uncertainties on the basis of multiple simulations.  It might make sense to fold the relevant sections together.

**Comment by mangiapasta** [comment]
I merged the Bootstrap / synthetic data section.  

Below I've written some general comments / thoughts on this section.  I'm happy to make changes now, let others do it, or hold off until a later stage, but I wanted to get my thoughts down now.

1) Title: I would change "Computing error in specific observables" to "Computing uncertainty in specific observables."  

2) If we use a standard document / set of documents as a source of definitions, we should state that upfront.

3) In that vein, I would changes the first instances of "standard error" to "standard uncertainty"

4) In the bootstrap section,  I added Andrew's example about the pitfalls of using a Gaussian to model free energies.  It would be good if someone gives that a thorough look-over to make sure I didn't say anything stupid (I don't typically work with free-energies).

**Comment by dmzuckerman** [comment]
@mangiapasta - thanks for all that.  some replies ...

1. Title: I would change "Computing error in specific observables" to "Computing uncertainty in specific observables."
** seems fine
2. If we use a standard document / set of documents as a source of definitions, we should state that upfront.
** do you have a suggestion for this?
3. In that vein, I would changes the first instances of "standard error" to "standard uncertainty"
** i think standard error (of the mean) is well known.  i'm not familiar with standard uncertainty ... are you trying to get at the difference between a confidence interval and the std err?  certainly that is important!
4. In the bootstrap section, I added Andrew's example about the pitfalls of using a Gaussian to model free energies. It would be good if someone gives that a thorough look-over to make sure I didn't say anything stupid (I don't typically work with free-energies).
** hopefully @ajschult can have a quick look

**Comment by ajschult** [comment]
@mangiapasta @dmzuckerman yes, the bit about applying the parametric bootsrapping to free energy looks good to me.

**Comment by agrossfield** [comment]
“Standard error” is the technical term from statistics

___________________________________
Dr. Alan Grossfield
Associate Professor
Department of Biochemistry and Biophysics
University of Rochester Medical Center
610 Elmwood Ave, Box 712
Rochester, NY 14642
Phone: 585 276 4193<tel:585%20276%204193>
http://membrane.urmc.rochester.eduu<http://membrane.urmc.rochester.edu>


On Oct 3, 2017 at 5:35 PM, <dmzuckerman<mailto:notifications@github.com>> wrote:


@mangiapasta<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mangiapasta&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=80W-rP8APCmwQIKc2ALT7vO1lxfPp5Y5Hv1OXv7iCZk&s=yK7g0SBJ5mgHblL7TaB4EFpXfMZXHeiwNKi5G55WSsg&e=> - thanks for all that. some replies ...

  1.  Title: I would change "Computing error in specific observables" to "Computing uncertainty in specific observables."
** seems fine
  2.  If we use a standard document / set of documents as a source of definitions, we should state that upfront.
** do you have a suggestion for this?
  3.  In that vein, I would changes the first instances of "standard error" to "standard uncertainty"
** i think standard error (of the mean) is well known. i'm not familiar with standard uncertainty ... are you trying to get at the difference between a confidence interval and the std err? certainly that is important!
  4.  In the bootstrap section, I added Andrew's example about the pitfalls of using a Gaussian to model free energies. It would be good if someone gives that a thorough look-over to make sure I didn't say anything stupid (I don't typically work with free-energies).
** hopefully @ajschult<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_ajschult&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=80W-rP8APCmwQIKc2ALT7vO1lxfPp5Y5Hv1OXv7iCZk&s=zqivywIWyiFxBQibbZ9_oFbCKBcSwLfvFUVekEu1-yY&e=> can have a quick look

**Comment by mangiapasta** [comment]
Re: documents for source of definitions. Dan Siderius had a pretty comprehensive list on the thread about terminology / NIST internal review.  I attached the "Guide to the expression
of uncertainty in measurement," which is one of our references for issues like this.  

So, for example, standard uncertainty is defined on page 3 as "uncertainty of the result of a measurement expressed as a standard deviation."   Note 3 on p. 36 states, '“Experimental standard deviation of the mean” is sometimes incorrectly called standard error of the mean.'


[JCGM_100_2008_E (1).pdf](https://github.com/dmzuckerman/Sampling-Uncertainty/files/1356084/JCGM_100_2008_E.1.pdf)

**Comment by dmzuckerman** [comment]
@mangiapasta @ajschult @agrossfield I would like to get your input on something sorta important ...

Currently the draft says, "When reporting these estimates, it is important to also provide an estimate of the uncertainty of the result, typically as a standard error."  On the one hand, this is perfectly reasonable, and once you know what is being reported, you essentially have all the info you need ... but on the other hand, I feel that folks will confuse this single std err with a healthy confidence interval - which it is not.  I'm not optimistic that our readers (and equally importantly, the readers of their papers) will convert from uncertainty scale to confidence interval in their minds.  And anything shown visually (like error bars!) leaves a lasting impression regardless of any more careful wording that may be in a paper.  So I would say we should encourage folks to show plus/minus TWO standard errors (std uncertainties).

All this was recently brought home to me when we were doing some validation studies comparing two data sets that should have agreed within noise, but the error bars only overlapped when we used two standard errors.

I wanted to get feedback on this before I edit the document.  If folks agree (or even if not) I can make a figure from the data I mentioned to illustrate the issue.

**Comment by mangiapasta** [comment]
I would actually argue a slightly different perspective, namely that the concept of a "healthy" confidence interval can't be divorced from a particular application or conclusion that one is trying to draw from the data.


I think degenerate cases illustrate the point.


If I'm trying to design an airplane wing using simulation, then my uncertainty bounds had better be conservative.  I want to be 99.999% sure that the wing isn't going to break in flight.  So I might pick a max-min confidence interval, or perhaps 5-sigma.  Admittedly this may be an unrealistic application for MD simulation, but I seem to recall that in astronomy, for example, 5-sigma is the minimum uncertainty bound that the community accepts as the basis for concluding that a measurement is statistically significant.


On the conservative side, I find Chebyshev's inequality useful for deciding what is a lower bound on uncertainty associated with a given sigma value.  See, e.g. the first few paragraphs of https://en.wikipedia.org/wiki/Chebyshev%27s_inequality .  The 68-95-99.7 rule is only valid for Gaussian distributions, which do not apply to all physical phenomena.


On the other hand, if one is only trying to elicit trends in data, one sigma may be perfectly reasonable.  I've done in material design problems where we only care about rank-ordering properties of different materials.


If I'm just reporting a generic uncertainty, e.g. for archival data, I usually go with 3-sigma.  By Chebyshev's inequality, that corresponds roughly to a 90% confidence interval.


________________________________

**Comment by dmzuckerman** [comment]
@mangiapasta Good points - especially about the purpose driving the approach.

Part of me would love to advocate for 3-sigma, but in the biomolecular world, that might invalidate all conclusions ever reported!  (Only half joking.)  In any case, we can explain the inequality issue.

FYI in the biomolecular world, a common goal is to compare two 'conditions' (a protein bound to a ligand or not; a wild-type vs a mutant protein; high [salt] vs low; ...) so conservative confidence intervals are indeed needed to enable confident distinctions.

@agrossfield - I'd love to get your view when you have the time.  This is our chance to set a meaningful standard for our field.  At a minimum we need to explain the issue carefully, I think.

**Comment by agrossfield** [comment]
I’ll write more when I have time, but part of me would be happy if people actually reported anything meaningful in terms of standard error.  The most common thing _still_ is just the standard dev from the sim itself, which is worthless as a convergence measure.  If someone reports +/- 1 se (e.g. computed from multiple trajectories), I’m happy, because I can mentally double it and see the difference.

I think it’s much more important to focus on getting people to actually compute a meaningful standard error, rather than a wish-it-was standard error.  :)

Alan

On Oct 11, 2017, at 1:53 PM, dmzuckerman <notifications@github.com<mailto:notifications@github.com>> wrote:


@mangiapasta<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mangiapasta&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=VpvMFjvzqB2RlxxaUG0bKNS47SJZzvLPU4ej8Gtavro&s=sng5dX7JA_wIu5w3wxwy_bSxd9S5bp7A18p48wQjYCA&e=> Good points - especially about the purpose driving the approach.

Part of me would love to advocate for 3-sigma, but in the biomolecular world, that might invalidate all conclusions ever reported! (Only half joking.) In any case, we can explain the inequality issue.

FYI in the biomolecular world, a common goal is to compare two 'conditions' (a protein bound to a ligand or not; a wild-type vs a mutant protein; high [salt] vs low; ...) so conservative confidence intervals are indeed needed to enable confident distinctions.

@agrossfield<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_agrossfield&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=VpvMFjvzqB2RlxxaUG0bKNS47SJZzvLPU4ej8Gtavro&s=2iaSUwbedU7fHFoVvvUEpAc6-0sbhyqn1hELEcwgB68&e=> - I'd love to get your view when you have the time. This is our chance to set a meaningful standard for our field. At a minimum we need to explain the issue carefully, I think.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_dmzuckerman_Sampling-2DUncertainty_issues_6-23issuecomment-2D335894340&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=VpvMFjvzqB2RlxxaUG0bKNS47SJZzvLPU4ej8Gtavro&s=g8D7FHk74z0rRALc8ylTvCrlTSaTGn8EDi3BNmXM5r0&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AM-5F-2D8o972U8W4-2DJfYNtbgoam94r-2DkRfXks5srQCYgaJpZM4PnxtI&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=VpvMFjvzqB2RlxxaUG0bKNS47SJZzvLPU4ej8Gtavro&s=cDNkBqOtfQBA0RNmwL17T_uEEAbbndHAtyJi2Acnd9o&e=>.

___________________________________
Dr. Alan Grossfield
Associate Professor
Department of Biochemistry and Biophysics
University of Rochester Medical Center
610 Elmwood Ave, Box 712
Rochester, NY 14642
Phone: 585 276 4193
http://membrane.urmc.rochester.edu

**Comment by ajschult** [comment]
I prefer reporting 1 sigma as the uncertainty.  I think of the error bars as a indication of the variability of the data rather than a bound on the true value (because there is no bound).  1 sigma is a far better description of the variability than 3 or 5 sigma.  Beyond that, conservative error bars just create a different problem.  Someone looks at two values (say 6 +/- 4 and 9 +/- 4) and says they agree because they're within each other's error bars.  But they actually disagree because sigma=1 and "4" is actually 4 sigma.

I do think it would be helpful to provide more explanation about confidence limits and how they relate to sigma.  We currently only mention it in passing as part of the "Best Practices" checklist.  And we can say that reporting something other than 1 sigma is an option, but that you should always explain what the reported uncertainties are (1 sigma, 3 sigma, etc).

**Comment by agrossfield** [comment]
Beautifully said, Andrew.

___________________________________
Dr. Alan Grossfield
Associate Professor
Department of Biochemistry and Biophysics
University of Rochester Medical Center
610 Elmwood Ave, Box 712
Rochester, NY 14642
Phone: 585 276 4193<tel:585%20276%204193>
http://membrane.urmc.rochester.eduu<http://membrane.urmc.rochester.edu>


On Oct 11, 2017 at 7:21 PM, <Andrew Schultz<mailto:notifications@github.com>> wrote:


I prefer reporting 1 sigma as the uncertainty. I think of the error bars as a indication of the variability of the data rather than a bound on the true value (because there is no bound). 1 sigma is a far better description of the variability than 3 or 5 sigma. Beyond that, conservative error bars just create a different problem. Someone looks at two values (say 6 +/- 4 and 9 +/- 4) and says they agree because they're within each other's error bars. But they actually disagree because sigma=1 and "4" is actually 4 sigma.

I do think it would be helpful to provide more explanation about confidence limits and how they relate to sigma. We currently only mention it in passing as part of the "Best Practices" checklist. And we can say that reporting something other than 1 sigma is an option, but that you should always explain what the reported uncertainties are (1 sigma, 3 sigma, etc).

**Comment by dwsideriusNIST** [comment]
All,
I've stayed out of this thread for various reasons. To get a digest version, can you clarify if the group is leaning toward declaration of a "error bars mean X (and only X)" statement in the paper?

My issue is that error bars are meaningless in the absence of stated terminology (see https://github.com/dmzuckerman/Sampling-Uncertainty/issues/9)

**Comment by mangiapasta** [comment]
Fair point.  It seems that we're conflating several issues here (or perhaps anticipating that junior scientists will...?).

Based on the previous comments, it seems to me that we are conflating the choice of error bar (e.g. 1-sigma, 2-sigma, etc.) with an assessment of statistical significance.  That is, if two data-points have overlapping error bars, are they statistically significant, or is there a difference?  One can always 
change the error bars so that there is much overlap or none at all.  So, in that sense, error bars are not statements about statistical comparison per se.  Maybe this is what Andrew was saying earlier.

On the other hand, Dan's comment is consistent with Alan's perspective (I think -- @agrossfield , chime in).  A lot of time when we read papers, we take error bars as a proxy for statistical comparison.  The latter can in fact be a complicated task because one has to take into account what type of uncertainty, should we interpret standard deviations using the 68-95-99.7 rule or Chebyshev's inequality, etc.  If we don't take the time to consider these issues, we may be misled by the error bars.  This is probably more of a problem for junior readers. 

So, @dwsideriusNIST I think we're trying to answer exactly your question.  Do we take error bars to always be 1-sigma because error bars are just a convention to represent uncertainty?  Or, do we suggest that error bars should reflect the underlying statistical comparison that we are trying to make with the uncertainty quantification?  

@dmzuckerman Is this the question you were asking?

**Comment by agrossfield** [comment]
I agree with what you wrote, but my point was even simpler. If we’re going to introduce a new term (standard uncertainty) to replace an earlier one (standard error), which is itself often used wrong, we have to not only tell people what the correct way to say it is, but also what is often meant when using the term incorrectly, especially because a lot of papers are not clear about what they did.

Said another way: we need to equip people to contribute to literature in the best way, while also helping them digest the existing literature, warts and all.

___________________________________
Dr. Alan Grossfield
Associate Professor
Department of Biochemistry and Biophysics
University of Rochester Medical Center
610 Elmwood Ave, Box 712
Rochester, NY 14642
Phone: 585 276 4193<tel:585%20276%204193>
http://membrane.urmc.rochester.eduu<http://membrane.urmc.rochester.edu>


On Oct 12, 2017 at 11:40 AM, <mangiapasta<mailto:notifications@github.com>> wrote:


Fair point. It seems that we're conflating several issues here (or perhaps anticipating that junior scientists will...?).

Based on the previous comments, it seems to me that we are conflating the choice of error bar (e.g. 1-sigma, 2-sigma, etc.) with an assessment of statistical significance. That is, if two data-points have overlapping error bars, are they statistically significant, or is there a difference? One can always
change the error bars so that there is much overlap or none at all. So, in that sense, error bars are not statements about statistical comparison per se. Maybe this is what Andrew was saying earlier.

On the other hand, Dan's comment is consistent with Alan's perspective (I think -- @agrossfield<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_agrossfield&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=RZJye_dKn5al-zY5RyWaHMhG4vRuM9Ng3nfoEGPQAHE&s=Oe0urxFSIiOF_nMNMVGny3eXE5rCvT_B7RHyC_pYlxs&e=> , chime in). A lot of time when we read papers, we take error bars as a proxy for statistical comparison. The latter can in fact be a complicated task because one has to take into account what type of uncertainty, should we interpret standard deviations using the 68-95-99.7 rule or Chebyshev's inequality, etc. If we don't take the time to consider these issues, we may be misled by the error bars. This is probably more of a problem for junior readers.

So, @dwsideriusNIST<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_dwsideriusnist&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=RZJye_dKn5al-zY5RyWaHMhG4vRuM9Ng3nfoEGPQAHE&s=zcHn6NA_5qXXRjoZj2ih4FKd8Gi8kEHnUghiYadATHU&e=> I think we're trying to answer exactly your question. Do we take error bars to always be 1-sigma because error bars are just a convention to represent uncertainty? Or, do we suggest that error bars should reflect the underlying statistical comparison that we are trying to make with the uncertainty quantification?

@dmzuckerman<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_dmzuckerman&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=RZJye_dKn5al-zY5RyWaHMhG4vRuM9Ng3nfoEGPQAHE&s=Bh43BLO_a3UgZtg0J6YQgCJ5OyX2mqP73ixaKiTyg-g&e=> Is this the question you were asking?

**Comment by dmzuckerman** [comment]
I agree with @agrossfield that we need to be careful with terms and err on the side of being overly explanatory and repetitive.  I don't think it's sufficient to state the definition once and assume readers will get it.

> @mangiapasta wrote "Do we take error bars to always be 1-sigma because error bars are just a convention to represent uncertainty? Or, do we suggest that error bars should reflect the underlying statistical comparison that we are trying to make with the uncertainty quantification?
@dmzuckerman Is this the question you were asking?"

Yes, more or less.  I'm thinking about statistical comparisons (both in a given paper and perhaps performed post-facto by another simulator or experimentalist) and people's understanding.  We have to assume that a good chunk of the ultimate consumers of reported error bars (i.e., our readers AND THEIR READERS) will not be energetic or sophisticated enough to extrapolate from one sigma as trivial as that may seem to us.  

So my own initial thought would be to recommend reporting 2*sigma (and also to say so!) ... and let the experts extrapolate back to one sigma if they like.   The experts will always be ok.  I worry about subtly promoting the continued mis-use of simulation data by giving our official blessing to misleadingly small error bars.

**Comment by mangiapasta** [comment]
@dmzuckerman 

> "So my own initial thought would be to recommend reporting 2*sigma (and also to say so!) ... and let the experts extrapolate back to one sigma if they like. The experts will always be ok. I worry about subtly promoting the continued mis-use of simulation data by giving our official blessing to misleadingly small error bars."

I think you and Alan are correct to point out the social engineering aspects of this document.  So I'd agree with your recommendation, provided we emphasize the distinction between error bars and statistical comparison.  To the extent that we can point readers to the fact that there is a "higher-level" of thinking (that sounds really hoity-toity ... sorry) with regards to statistical comparison, I think we only help the community.   This emphasis need not be overly drawn out, but I do think it would be good to mention some of the issues we've written about in this thread.

**Comment by dmzuckerman** [comment]
From today's discussion
- Section to be renamed 'Error bars and uncertainties in specific observables'
- Must introduce two ideas: scale of uncertainty (std uncert) and confidence interval
- Recommend reporting conf intervals based on t-test
- Give caveats: different confidence intervals for different purposes; also not specific tests for assessing whether two estimates agree or disagree
- INCLUDE COMMON ERRORS!!  (esp std dev vs std uncert/error)
- Use standard terminology and hyperlink
- Jack-knife should be mentioned for small data sets
- Small data sets (N<5): just show all values.

**Comment by dmzuckerman** [comment]
@dwsideriusNIST @mangiapasta @ajschult @agrossfield Today I really sweated and updated the specific observables section following our discussion ... to recommend using 90% confidence intervals.  I think this is the most important section of the paper (after the definitions, of course, and the quick-and-dirty checks which will eliminate 95% of studies) so please read it when you have time and critique/edit.  Thank you!  --Dan

**Comment by dmzuckerman** [comment]
Paul @mangiapasta, would you please see whether you agree that the section you drafted (7.3, now called 'Assessing qualitative behavior of data') should be moved to 'quick and dirty' section?

Also, I would appreciate your review of the sections I revised/added
- 7.1 'Basics'
- 7.6 'From standard uncertainty to confidence interal for Gaussian variables'

Thank you!

**Comment by mangiapasta** [comment]
I'll take a look today and get back to you soon.

________________________________

**Comment by mangiapasta** [comment]
I'm fine with merging the "qualitative behavior of data" section with the quick and dirty section.  But can we perhaps rename the latter?  "Quick and dirty" sounds a little pejorative, like it shouldn't be taken too seriously.  Maybe something like "Preliminary checks that can rule out..." or similar?


________________________________

**Comment by dmzuckerman** [comment]
@mangiapasta you make a very good point - I agree folks might not want to invest time into something quick and dirty!

What if we call it "Essential Qualitative Analysis" or "Essential Preliminary Analysis"?  I like having the word 'essential' (or similar) in there to indicate the importance.

**Comment by mangiapasta** [comment]
I'm fine with some variation on that.  As an alternative, you might also consider something like "Initial data screening" or something similar.  I like the word screening if only because it suggests filtering out bad data, but (hopefully) in a systematic way

________________________________

**Comment by mangiapasta** [comment]
Overall I like section 7.1 but thought it needed a bit of reorganization.


Basically, it seems to me that there are two main issues you are trying to address.


First, why should we use confidence intervals instead of standard uncertainties?  It sounds like you there are two related answers to this

A) Confidence intervals are more relatable to everyday experience because it is a frequentist approach to uncertainty (i.e. we expect an outcome 90% of the time)

B) Confidence intervals don't suffer from under (or over) estimating uncertainty.  Admittedly, I added the part about over estimating, but I think that can be as much of a sin as underestimating.


Second, why are we emphasizing the issue of how to report error bars so much up front.


Here it seems that your argument is that we are dealing with the intersection of a societal and scientific issue.  The main points seems to me to be that (i) error bars are necessary to accurately assess work, and (ii) even accurate but (visually) misleading error bars can be detrimental if readers aren't careful.


Is this an accurate summary of what you had in mind?  I reorganized the text to (hopefully) conform to this structure.  I pasted it below instead of making changes in the main file.  Let me know your thoughts and/or edit as appropriate.  I'll look at the other section tonight/tomorrow.


Also, I'd like to drop the word "reproducibility" in here because I think that is one of the goals of UQ, namely to indicate the level of reproducibility of a result.


Paul






``What error bar should I report?''
Here we address this simple but critical question.

In general, there is no one-best practice for choosing error bars.  However, in the context of simulations, we can nonetheless identify common goals when reporting such estimates: (i) to help authors and readers better understand uncertainty in data; and (ii) to provide readers with realistic information about the reproducibility of a given result.

With this in mind, we recommend the following:
(a) in fields where there is a definitive standard for reporting uncertainty, the authors should follow existing conventions;
(b) otherwise, such as for biomolecular simulations, \emph{authors should report (and graph) their best estimates of 90\% confidence intervals.}
As explained in the glossary above, a 90\% confidence interval is a range of values that is expected to bracket 90\% of the computed predictions \emph{if statistically equivalent simulations are repeated a large number of times.}

We emphasize that as opposed to standard uncertainties (reported as a standard deviation $\sigma$), confidence intervals have several practical benefits that justify their usage.  In particular, they directly quantify the statistical frequency with which we expect a given outcome, which is more relatable to everyday experience than moments of a probability distribution.  As such, confidence intervals can help authors and readers better understand the implications of an uncertainty analysis.  Moreover, downstream users/readers of a given paper may include less statistically-oriented readers for whom confidence intervals are a more meaningful measure of variation.

In a related vein, error bars expressed in terms of $n$ $\sigma$ can be misinterpreted as unrealistically under or overestimating uncertainty if taken at face value.  For example, reporting $3$ $\sigma$ uncertainties for a normal random variable amounts to a $99.7$ \% confidence interval, which is likely to be a significant overestimate for many applications.  On the other hand, $1$ $\sigma$ uncertainties only correspond to a $68$ \% confidence interval, which may be too low.  Given that many readers may not take the time to make such conversions in their heads, we feel that it is safest for modelers to do this work up front.

In recommending 90 \% confidence intervals, we are admittedly attempting to address a social issue that nevertheless has important implications for science as a whole.  In particular, the authors of a study and the reputation of their field do not benefit in the long run by under-representing uncertainty, since this may lead to incorrect conclusions.  But perhaps just as importantly, many of the same problems can arise if uncertainties are reported in a technically correct but obscure and difficult-to-interpret manner.  For example, 1 $\sigma$ error bars may not overlap and thereby mask the inability to statistically distinguish two quantities, since the corresponding confidence intervals are only 68 \%.  With this in mind, we therefore wish to emphasize that visual impressions conveyed by figures in a paper are of primary importance.  Regardless of what a research paper may explain carefully in text, error bars on graphs create a lasting impression and must be as informative and accurate as possible.  If 90\% confidence intervals are reported, the expert reader can easily estimate the smaller standard uncertainty (especially if it is noted in the text), but showing a graph with overly small error bars is bound to mislead most readers -- even experts who do not search out the fine print.





________________________________

**Comment by mangiapasta** [comment]
Overall I think section 7.6 is fine.  The only change I made was to edit one of the table entries, which was originally k=1.03.  I think it should have been 1.73 interpolating between the two values surrounding it.  Would be good to check that I put the correct value.  Otherwise the section reads well to me.  I think it's fine to suggest that folks show data for 5 or less values and start using the t-distribution for 6 or more data points.

**Comment by dmzuckerman** [comment]
@mangiapasta thanks a lot.  I looked up that number - happens to be 1.72 - and fixed it.  Good catch.

**Comment by dmzuckerman** [comment]
@mangiapasta somehow I missed that whole revision of Sec. 7.1 in this thread!  I think your clarifications/edits are excellent.  The one quibble I have is with this wording: "standard uncertainties (reported as a standard deviation $\sigma$)."  Since we are (I think) referring to what our readers will know as the std err of mean, we should say so.  A lot of readers won't have the statistical sophistication to appreciate that the std err is just a std dev (for the mean) and so saying 'reported as a std dev' will be very confusing.  If we don't say it's the std err, perhaps just omit further explanation.  I had said something about the scale of uncertainty, but that may also not make sense to some ... it's physics-y jargon.

Anyway please go ahead and push your edits to the master.  Thanks a lot for doing that so thoughtfully.

**Comment by mangiapasta** [comment]
Sounds good.  I'm fine with the changes you suggest.  I'll push edits to the master today

________________________________

**Comment by mangiapasta** [comment]
@dmzuckerman 
 
I read your comment more carefully and wanted to follow up on the main point.  It's not clear to me that we're interested in standard error of the mean every time we report an error bar or confidence interval.  That's why I originally "standard uncertainties (reported as a standard deviation $\sigma$)."  I think Andrew Schultz at one point mentioned the perspective that error bars can be useful for showing spread in data, which would just be the standard deviation of the data itself.  What are your thoughts?

For the time being, I've left the revision as is but will go in and change the language if appropriate.

Thanks

**Comment by dmzuckerman** [comment]
@mangiapasta I moved your subsection on qualitative behavior (of strain) to the Qualitative (formerly quick-and-dirty) section.  Please check whether you're happy with it as written in that location.  Thank you.

**Comment by dmzuckerman** [comment]
@mangiapasta sorry for all the notes - this should be the last one.  I was thinking we should probably omit the paragraph and table you wrote about significant figures at the end of Sec. 7.1, though perhaps it can be estimated to be consistent with our current approach and terminology.  See what you think.

**Comment by mangiapasta** [comment]
I didn't write that; not sure where it came from.  Feel free to do what you like with that.

________________________________

**Comment by dmzuckerman** [comment]
@dwsideriusNIST @ajschult did one of you make this table at end of Sec. 7.1 and write the accompanying text?  I was thinking we should omit it or revise to be consistent with rest of paper.  Thank you.  (Sorry @mangiapasta about mistaken authorship.)

![image](https://user-images.githubusercontent.com/21345239/33686430-c069027c-da89-11e7-8253-d1a64890a4fc.png)

**Comment by dwsideriusNIST** [comment]
I'm not sure where that came from...

**Comment by ajschult** [comment]
Right, I wrote that based on the item in our planning document about "basics" (see comments in this issue from Sep 30, Oct 1).  I think when we were in D.C. that someone mentioned people reporting values with absurd numbers of digits.  In any case, I'm not emotionally attached to the paragraph; parts of it certainly seem to be redundant with the rest of section 7.1

**Comment by dmzuckerman** [comment]
Thanks @ajschult .  I think the point about absurd numbers of digits is valid - and the problem can contribute to misunderstanding.  Would you be willing to rewrite this bit so that it's a narrow point about significant figures aligned with uncertainty?  I think the table can probably be omitted unless it will help to make the point.  Thank you.

**Comment by mangiapasta** [comment]
So I actually broke up the subsection I wrote in the qualitative behavior section.  In particular, I moved the first paragraph to the beginning of the entire section, and then recast the rest of my subsection as an example on how to determine the existence of linear regimes in data.  I think overall the quick-and-dirty section is really about assessing quality of data, and we provide examples of various ways to do this.  Feel free to change / edit if you don't think that's appropriate.

________________________________

**Comment by dmzuckerman** [comment]
I'll look at this after Sec. 7 is revised so I can do everything at once.

**Comment by dwsideriusNIST** [comment]
I pushed some updates to specific observables, fleshing out the block averages section and adding one on autocorrelation analysis. Definitely not yet, but on the way.

I think the intro to sec 7 is going to need a few "why we recommend 90% CI" sentences, because a lot of readers/reviewers are going to be surprised that 95% is not recommended straightaway. The thing is, I don't think that there is anything special about 95%. It probably became a de facto standard because the +/- ~2s confidence interval makes for easy math. Without some statement of "why" we open up the accusation of cooking the data to look better.

**Comment by dmzuckerman** [comment]
Good point, @dwsideriusNIST: "readers/reviewers are going to be surprised that 95% is not recommended"

I'm probably guilty of suggesting 90% but not for a good reason.  I'm happy enough if we change to 95%.

Should we?  @mangiapasta ? @agrossfield ?

**Comment by dwsideriusNIST** [comment]
95% will be a safe choice if we want to avoid criticism, founded or unfounded.

Ultimately, we need to stress that the writer should report the chosen confidence level. No one *should* argue with that; as a reviewer I would frown on someone using a 50% confidence level (OK, it's absurd, but just tag along for a moment) to set error bars, but if they say that honestly and upfront, I'll accept it. I'll be more critical of claims based on the CI, but the statistics are at least honest.

**Comment by mangiapasta** [comment]
I'm fine with whatever confidence intervals we end up suggesting.

________________________________

**Comment by dmzuckerman** [comment]
@dwsideriusNIST and @mangiapasta - I changed to 95%ile CI recommendation and updated corresponding table for coverage factors.  Thanks again for suggesting that.

**Comment by dwsideriusNIST** [comment]
@mangiapasta @dmzuckerman I think we may need to make a change of notation to define the arithmetic mean and true value.

In the specific observables section, we need a functional representation of the arithmetic mean; currently we use an overbar, but I don't think this is going to work well. E.g., equation 14 on page 13. We may need to use the <...> notation for arithmetic means and something else for the true value.

This is a departure from the VIM, but the paper needs to express arithmetic means of mathematical expressions whereas the VIM doesn't address that level of computation.

One option: Use <x> to indicate the arithmetic mean of x. Use E[x] to indicate the true expectation value.

I'm open to other suggestions. One benefit is that <x> is more familiar to simulators.

**Comment by dmzuckerman** [comment]
@dwsideriusNIST and @mangiapasta some brief comments:
- in eq (11), can you just use an overbar?  i think this ends up being a sample avg - just explain how it's done.  also, isn't s(q_k) just s(q)?  some clarification on this.
- in eq (12) I don't think N_lags is defined
- I find the section on parametric bootstrapping to be technical and difficult to understand (for me).  For example, is the 'deterministic function' a distribution and how is it obtained?  Either this discussion should be significantly revised and clarified with a realistic example or it should be omitted.  We can just make mention of the approach and give a reference.

**Comment by dmzuckerman** [comment]
@dwsideriusNIST thanks.  Regarding notation, I'm not certain what the best thing is.  In statistical mechanics the < . > notation usually refers to a true expectation.  The overbar in stat mech is usually defined in context, so a sample average seems should be an ok way to use it.

I do have the worry that the E[.] notation will be off-putting to some, even though it shouldn't.  Anyone else have thoughts before Dan S does a bunch of work on this?  @drroe @agrossfield ?

**Comment by dwsideriusNIST** [comment]
I would definitely appreciate any thoughts regarding the mean/expectation notation. Please speak up if equation 11 is OK; it's only an aesthetic issue for me.

As a side note, the MC community uses the <.> notation for both pencil/paper expressions for the theoretical expectation value and a simulation-obtained ensemble average.

**Comment by dwsideriusNIST** [comment]
I decided to try out the E[x] and <x> notation in a feature branch. Please check the main.pdf file in the present commit of the avg_notation branch to see its effect: https://github.com/dmzuckerman/Sampling-Uncertainty/commit/5a68effdc70089ae7f1f1c2c38b5c75ff8de9acb

In particular, compare eq. 11 on page 14 to that in the master branch.

**Comment by dmzuckerman** [comment]
@dwsideriusNIST thanks for trying that.  In the end, I prefer <.> and overbar, not E[x].  I think the one that decided me was Eq (7), where you didn't yet put the E[ . ] around the whole thing ... but that gets particularly cumbersome.

**Comment by dwsideriusNIST** [comment]
Good catch... OK, there are only votes against the proposed notation, so I'll abandon that branch.

I am going to bring some macros from the avg_notation branch over to the main branch. (I created \expval and \mean macros that allowed me to switch notation quickly.) Might add one or two for the standard uncertainties as well as it helps enforce consistency.

**Comment by agrossfield** [comment]
I like the macros idea — I have a ton of macros I use for latex, but I’ve refrained from introducing them here because I didn’t want to impose my way on everyone else.

Cheers,

Alan


On Jan 5, 2018, at 1:33 PM, Daniel W. Siderius <notifications@github.com<mailto:notifications@github.com>> wrote:


Good catch... OK, there are only votes against the proposed notation, so I'll abandon that branch.

I am going to bring some macros from the avg_notation branch over to the main branch. (I created \expval and \mean macros that allowed me to switch notation quickly.) Might add one or two for the standard uncertainties as well as it helps enforce consistency.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_dmzuckerman_Sampling-2DUncertainty_issues_6-23issuecomment-2D355630349&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=SnpxlY2nNA_K0OfPChDiMAlwTp99H_AATmp3m-eIrPs&s=d9wFltViHQQEgULsG47aZ1BkkFRHsbv1lj0R7YKqVg4&e=>, or mute the thread<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_notifications_unsubscribe-2Dauth_AM-5F-2D8od1mqh-5FkOcdVOG85DE3Ol-2Dft-2D7kks5tHmr4gaJpZM4PnxtI&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=SnpxlY2nNA_K0OfPChDiMAlwTp99H_AATmp3m-eIrPs&s=qj_0lUwEw-9idWU97w6fXUK7KnKhR5vh8Wg-c_-CJPw&e=>.

___________________________________
Dr. Alan Grossfield
Associate Professor
Department of Biochemistry and Biophysics
University of Rochester Medical Center
610 Elmwood Ave, Box 712
Rochester, NY 14642
Phone: 585 276 4193
http://membrane.urmc.rochester.edu

**Comment by dwsideriusNIST** [comment]
@agrossfield thanks for the vote of confidence. The most recent commit has a few macros, now used in *most* parts of the paper (dark uncertainty, propagation of uncertainty currently excluded):
\expval{x}: the expectation value of x
\mean{x}: the arithmetic mean of x
\stdev{x_k}: the experimental standard deviation of x, given a set of measurements {x_k}
\stdevmean{\mean{x}}: the experimental standard deviation of the mean of x

(currently, \stdev and \stdevmean yield the same, but I kept them separate in case we wanted to use different notation for the \stdevmean)

As the paper gets closer to finality, I'll continue to convert any residual statistical quantities to use these macros, so far as is possible.

**Comment by mangiapasta** [comment]
I'm part way through the specific observables section.  I left many comments in the .tex file.  @dmzuckerman et al., any feedback on the questions would be useful.  I can incorporate changes while I edit if you're able to look over them in the next few days.

**Comment by dmzuckerman** [comment]
I looked at your comments, @mangiapasta .  One issue is this section has been gone over by a few folks, so there is some heterogeneity in the writing ... both for style and content.  However, ultimately we need a consistent (and correct) document.  I trust your instincts on cleaning things up.  Here's how I suggest you proceed:
- Clarity (case I): When you understand and agree with the intent of a bit of writing but find it may be unclear for readers, please edit to improve clarity, precision.  I think you've been doing this already.
- Clarity (case II): When you simply don't understand something or fear it may be wrong, just comment it out with % but leave it there so others can adjust easily later if needed.
- Accuracy: If something is not correct, please correct it.  If you feel the best solution is omission, please comment it out.
- Redundancy: In general, I think redundancy isn't so bad for important points.  Also, I imagine many readers will selectively address their attention to specific sections, so repeating or summarizing content from another section is very appropriate - but of course, a cross-reference should be given.
- Gray area: If there's something you feel uncomfortable editing or deleting, just leave your comment in there and I will go through it all after.

**Comment by dwsideriusNIST** [comment]
@mangiapasta @dmzuckerman Current commit (https://github.com/dmzuckerman/Sampling-Uncertainty/commit/ac02c666e36d98c0113d33da502ed54f257bcf14) has some replies to PNP's comments. I cut the clause that seemed to imply that time correlation functions required bootstrapping.

**Comment by dwsideriusNIST** [comment]
Clicked the wrong "update"... sorry.

**Comment by dwsideriusNIST** [comment]
Can someone with more bootstrap experience clarify the section "Bootstrapping and uncertainty propagation?" To me, this section is specific to the parametric bootstrap.

For example, if I try to calculate the mean and uncertainty of, say \exp{-\beta U} using regular, nonparametric bootstrap and sampling from collected measurements of U (internal energy), then the resultant distribution of \exp{-\beta U} will obey the physical constraints (can't be negative) on that quantity by its mathematical nature.

If, however, I use a modelling approach in which we presume a distribution of \exp{-\beta U} but do not constrain it appropriately, then it's ripe to yield unphysical samples.

Or am I missing something else?

**Comment by ajschult** [comment]
@dwsideriusNIST Yes, that's correct.  That section could be folded into the previous section, but in with the bits about parametric bootstrapping.

**Comment by dwsideriusNIST** [comment]
@ajschult Thanks for the clarification. I'm working on a revision that movies that section to the "bootstrapping variants"

**Comment by dwsideriusNIST** [comment]
@mangiapasta Paul, I revised the error propagation section. It is significantly expanded, to give some background on 1) where the error propagation expression comes from and 2) why we use a Taylor-series expansion. Please give it a read to see if I'm off base anywhere. I did assume linearly uncorrelated raw data throughout, to avoid introducing covariances, etc., though with a reference that readers can follow if interested.

**Comment by mangiapasta** [comment]
Okay thanks.  I'll try to give this a look soon.  Currently we're in the middle of IMS season, with a large deadline coming up this Friday.  I can't promise anything before then, but I should be back on this by early next week at the latest.

**Comment by dwsideriusNIST** [comment]
I've added a PDF to the repo.

**Comment by dmzuckerman** [comment]
@dwsideriusNIST and @mangiapasta please check out my recent commit which is mostly quite small stuff in response to NIST reviewers.  One thing of substance I have realized since we started this project is that bootstrapping confidence intervals (CIs) are not completely reliable, so I added a cautionary note and a couple of references.  

And this is not in the manuscript, but ... I would probably go a little further to say that for some data sets (e.g., some of mine!) it is intrinsically impossible to make a reliable 95% CI because the sample is simply too small.  When you only have a small number, say O(10), of i.i.d. samples from a distribution which is unknown, there is always a decent chance that you have missed a value (part of the distribution) which will completely throw off your average.  Although you can estimate the chances that you missed such a value, you can't know how much it will change your estimated mean ... and hence you can't make a CI.  This is somewhat analogous to the chances of completely missing a free energy basin in a simulation of a physical system, but it's more mathematically fundamental because it arises even for i.i.d. samples.

Should we write up something about this issue??

Aside from that, I'm ready to submit.  Are we still waiting on NIST?

**Comment by agrossfield** [comment]
This is where I usually reference Rumsfeld talking about "unknown unknowns".

Dr. Alan Grossfield
Department of Biochemistry and Biophysics
University of Rochester Medical Center

On Apr 26, 2018, at 4:57 PM, dmzuckerman <notifications@github.com<mailto:notifications@github.com>> wrote:


@dwsideriusNIST<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_dwsideriusNIST&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=EdijO5AlYUPX4iic1QKVVJB0MFZyW8RqoYlGebdpd80&s=LkDv0NePgCpHpwDZgDwJ_nKbg7f7y21XBC37i7ZuUrc&e=> and @mangiapasta<https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_mangiapasta&d=DwMFaQ&c=4sF48jRmVAe_CH-k9mXYXEGfSnM3bY53YSKuLUQRxhA&r=49qnaP-kgQR_zujl5kbj_PmvQeXyz1NAoiLoIzsc27zuRX32UDM2oX8NQCaAsZzH&m=EdijO5AlYUPX4iic1QKVVJB0MFZyW8RqoYlGebdpd80&s=Tz2jea2E_vWNlcCvEdE7JIOaFDayIYt_HjlpIphrw4Y&e=> please check out my recent commit which is mostly quite small stuff in response to NIST reviewers. One thing of substance I have realized since we started this project is that bootstrapping confidence intervals (CIs) are not completely reliable, so I added a cautionary note and a couple of references.

And this is not in the manuscript, but ... I would probably go a little further to say that for some data sets (e.g., some of mine!) it is intrinsically impossible to make a reliable 95% CI because the sample is simply too small. When you only have a small number, say O(10), of i.i.d. samples from a distribution which is unknown, there is always a decent chance that you have missed a value (part of the distribution) which will completely throw off your average. Although you can estimate the chances that you missed such a value, you can't know how much it will change your estimated mean ... and hence you can't make a CI. This is somewhat analogous to the chances of completely missing a free energy basin in a simulation of a physical system, but it's more mathematically fundamental because it arises even for i.i.d. samples.

Should we write up something about this issue??

Aside from that, I'm ready to submit. Are we still waiting on NIST?

-

**Comment by dwsideriusNIST** [comment]
Dan, I'll look at your revisions either this afternoon (east coast time) or on Monday. I would guess we have 1 1/2 weeks max before NIST approval; it needs one more signature before going to to the review committee.

**Comment by mangiapasta** [comment]
I'm reading through the whole manuscript in detail now.  I'll get back to you in a day or so with comments.  Currently through 10 of 23 pages.  Overall this is looking much better.  I like the RMSD discussion now.  It needs a small amount of tweaking in my opinion, but nothing major.  I will weigh in on that when I compile my full list of comments.

________________________________

**Comment by dmzuckerman** [comment]
@mangiapasta please just edit (rather than commenting) unless you think something is very controversial.  This will save time.  I'll be happy to look over any changes from your commit.  Thanks!

**Comment by mangiapasta** [comment]
Will do

________________________________

**Comment by mangiapasta** [comment]
@dwsideriusNIST Dan, in Eqs. 18 and 19 of the Specific Observables section, there is weird notation _{x_{l \neq j}} appearing after one of the parentheses and before the larger vertical bar.  Is that a typo or did you mean something by it?  I couldn't figure out what that "l" (lowercase L) was referring to, and it didn't seem consistent with what you were trying to say re: Taylor's theorem. Also, in Eqs. 16 through 19 you use {x_i} to denote a collection of raw measurements.  The set notation throws me for a loop because sets are generally unordered objects.  I think it's more natural to write \boldsymbol {\roman x} = (x_1,x_2,...,x_M) so that, e.g. I can intrepret the RHS of Eq. 16 as a dot product.  Did you have a specific reason for using set notation?

**Comment by dwsideriusNIST** [comment]
@mangiapasta I looked over the equations to figure out what the issue is. The weird notation you pointed out was the "hold these fixed" part of the partial derivative - the "not equal to" sign was mangled, it must be a font issue. It's a notation convention that I (many of us) use in macroscopic thermodynamics, i.e., use an index variable (in this case "l") that is otherwise unused to indicate which generic variables are held fixed.

I'm OK dropping the "hold these fixed" list; it should be obvious that the partial derivative w.r.t. x_j is done while holding the other variables fixed.

Regarding the set notation, my original version of that section actually used vector/tensor notation (eq. 16 as "F = c + a.x", eq 18 as F = F(bar{x}) + grad(F).x + ... ; etc.). Eqs 17 and 19 start to get really nasty, as you have to introduce a transpose vector and then represent the variance as a variance-covariance tensor. In the end, I switched to set notation to avoid confusing readers unfamiliar with the tensor approach.

**Comment by mangiapasta** [comment]
Ah, okay I see what you were saying about the "hold these fixed".  I've seen similar things in some stat-mech textbooks, but never with the not-equal sign.  Anyways, I have a modest preference for removing that if you're okay with it.  The partial derivative implies the other variables should be fixed, as you point out.


It seems like you had a justification for using the set notation, so let's leave it unless someone in WERB (Andrew?) takes issue.

________________________________
